### PR TITLE
SOLR-16601: Remove Deprecated Methods from SolrClients

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-16601

# Description

Please provide a short description of the changes you're making with this pull request.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by epugh** [comment]
Hoping Crave.io doesn't get in a weird git state, and it can run the tests as I remove methods for me!

**Comment by epugh** [comment]
HttpSolrClient.setQueryParams is deprecated, but used ;-(  Need to migrate it...

**Comment by epugh** [comment]
Check logic around ConcurrentUpdateSolrClient and `setPollQueueTime`...

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java [hunk 1]
```diff
@@ -116,12 +114,23 @@ protected ConcurrentUpdateSolrClient(Builder builder) {
...
-    if (stallTime < pollQueueTime * 2) {
+    this.pollQueueTimeMillis = builder.pollQueueTime;
+    this.stallTimeMillis = Integer.getInteger("solr.cloud.client.stallTime", 15000);
+
+    // make sure the stall time is larger than the polling time
+    // to give a chance for the queue to change
+    int minimalStallTime = pollQueueTimeMillis * 2;
+    if (minimalStallTime > this.stallTimeMillis) {
+      this.stallTimeMillis = minimalStallTime;
+    }
```
Comment (janhoy) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java:125]: Should these changes, and the one below, be in a separate PR?

File: solr/core/src/test/org/apache/solr/cloud/TestRandomFlRTGCloud.java [hunk 2]
```diff
@@ -147,6 +148,7 @@ public class TestRandomFlRTGCloud extends SolrCloudTestCase {
           new NotIncludedValidator("score", "score_alias:score"));
 
   @BeforeClass
+  @SuppressWarnings({"rawtypes", "unchecked"})
```
Comment (risdenk) [inline solr/core/src/test/org/apache/solr/cloud/TestRandomFlRTGCloud.java:151]: why?

File: solr/core/src/test/org/apache/solr/cloud/TestRandomFlRTGCloud.java [hunk 3]
```diff
@@ -635,12 +644,36 @@ private static SolrDocumentList getDocsFromXmlResponse(
...
+  public static SolrClient getRandomClient(Random rand) {
+    int numClients = CLIENTS.size();
+    int idx = TestUtil.nextInt(rand, 0, numClients);
+    if (idx == numClients) {
+      // Return the CloudSolrClient, it only uses javabin writerType.
+      return COLLECTION_CLIENT;
+    } else {
+      // Grabbing the first
+      int rnd = random().nextInt(CLIENTS.get(idx).values().toArray().length);
+      return (SolrClient) CLIENTS.get(idx).values().toArray()[rnd];
```
Comment (risdenk) [inline solr/core/src/test/org/apache/solr/cloud/TestRandomFlRTGCloud.java:658]: Seems very convoluted... no need to create an array twice. Not even sure why need to create an array?

**Review by risdenk — CHANGES_REQUESTED** [review]

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java (hunk 1 above)
Comment (epugh) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java:125]: What I found was that the deprecated method setPollQueueTime is being used..  and that the logic iin the method needed to be preserved, so then I moved it...   I did want to have an extra set of eyes on it ;-).    WOuld you like me to cherry-pick the commit into a new PR???

**Comment by epugh** [comment]
@risdenk going to comment here on `TestRandomFlRTGCloud`.   

That test does some "interesting" things in randomizing things that is actually quite different then other tests (and indeed may have some things like it's use of various `FlValidator` classes that we may want to use elsewhere)..    The part I struggled with is that it randomly picks between `wt` JavaBin, Json, and XML, so I created SolrClients using the builder for each one.  Those can be pointed at specific nodes...     I just tweaked the Javadocs a bit to try and better explain:

```
   * returns a random SolrClient -- either a CloudSolrClient, or an HttpSolrClient pointed at a node
   * in our cluster.  We have different CLIENTS based on their wt setting for each node.
```

Honestly, I do find it a bit convoluted..    The test passes, but I don't love it.  If you have some ideas on how to make it less convoluted, I'd love that ;-)

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java (hunk 1 above)
Comment (janhoy) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java:125]: Ok, I saw that now. Looks good.

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java [hunk 4]
```diff
@@ -116,12 +114,23 @@ protected ConcurrentUpdateSolrClient(Builder builder) {
...
+
+    if (stallTimeMillis < pollQueueTimeMillis * 2) {
       throw new RuntimeException(
-          "Invalid stallTime: " + stallTime + "ms, must be 2x > pollQueueTime " + pollQueueTime);
+          "Invalid stallTime: "
```
Comment (janhoy) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java:129]: This condition will never happen, as we now force `this.stallTimeMillis` to at least 2x pollQueueTime. So just remove lines 127-133?

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java (hunk 1 above)
Comment (epugh) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java:125]: Awesome!

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java (hunk 4 above)
Comment (epugh) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java:129]: Thanks, did the same in the Http2 version..

File: solr/core/src/test/org/apache/solr/cloud/TestRandomFlRTGCloud.java (hunk 2 above)
Comment (epugh) [reply solr/core/src/test/org/apache/solr/cloud/TestRandomFlRTGCloud.java]: okay, looks like i got the right magic incantation.....

**Comment by epugh** [comment]
> @epugh I pushed [9d50904](https://github.com/apache/solr/commit/9d509044238297b0b2fe7bcb7aef946f3f3034f1) which simplifies things I think a lot to just compute a random client if it hasn't already been created yet.

I learned about "Pair"!   This is great, thansk...
